config/cron_udp.hjson

```diff
@@ -0,0 +1,364 @@
+                left join entity.learner_activity_result lar on (cae.short_assignment_id = lar.learner_activity_id and cae.short_user_id = lar.person_id )
+                left join keymap.learner_activity_result lar2 on lar.learner_activity_result_id  = lar2.id
+                left join keymap.person p on p.id = cae.short_user_id 
+            order by assignment_id, user_id
+        ),
+        all_assign_sub as 
+        (
+            select
+                cast(submission_id as BIGINT) AS id, 
+                assignment_id AS assignment_id, 
+                course_id, 
+                user_id, 
+                (
+                    case 
+                        when 
+                            (grade_posted_local_date is null or submission_workflow_state != 'graded') 
+                        then
+                            null 
+                        else 
+                            round(published_score,1) 
+                        end
+                ) AS score, 
+                submitted_at AS submitted_at, 
+                graded_at AS graded_date, 
+                grade_posted_local_date 
+            from 
+                course_assignment_submission 
+            order by assignment_id
+        )
+        select 
+            f.id::bigint, 
+            f.assignment_id::bigint, 
+            f.course_id, 
+            f.user_id, 
+            f.score::float, 
+            f.submitted_at, 
+            f.graded_date, 
+            f.grade_posted_local_date,
+            cast(f1.avg_score as float) as avg_score
+        from 
+            all_assign_sub f join 
+            (
+                select 
+                    assignment_id,
+                    round(avg(score),1) as avg_score
+                from 
+                    all_assign_sub 
+                group by assignment_id
+            ) as f1 
+            on f.assignment_id = f1.assignment_id
+        order by assignment_id, user_id
+        '''
+},
+# BigQuery event queries
+"RESOURCE_ACCESS_CONFIG": {
+    "canvas": {
+        "query": 
+            '''
+            SELECT 'canvas' AS resource_type, 
+            REGEXP_EXTRACT(object.id, r'.*:(.*)') AS resource_id, 
+            CAST(REGEXP_EXTRACT(membership.id, r'.*:(.*)') AS INT64) AS user_id, 
+            cast(null as string) AS user_login_name, 
+            CAST(REGEXP_EXTRACT(`group`.id, r'.*:(.*)') AS INT64) AS course_id, 
+            COALESCE( 
+            JSON_EXTRACT_SCALAR(object.extensions, '$[\'com.instructure.canvas\'][asset_name]'), 
+            JSON_EXTRACT_SCALAR(object.extensions, '$[\'com.instructure.canvas\'][filename]'), 
+            object.name, 
+            'attachment' 
+            ) as name, 
+            datetime(EVENT_TIME) as access_time 
+            FROM event_store.expanded 
+            where 
+            JSON_EXTRACT_SCALAR(ed_app.json, '$.id') IN UNNEST(['http://m.canvas.umich.edu/', 'http://umich.instructure.com/']) 
```

Comment:
We should probably look into making these configurable values

Reply:
That has been made as a new configuration variable "CANVAS_EVENT_URLS" into env.json file.